### PR TITLE
Checkout V2: Allow AVS and CVV for Authorizations

### PR DESCRIPTION
**Pull request by deedeelavinder**
AVS and CVV result details will also now come through on authorizations.

Remote Tests:
24 tests, 57 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions,
0 notifications
100% passed

Unit Tests:
19 tests, 80 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions,
0 notifications
100% passed

### CODE REVIEW

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 1]
```diff
@@ -132,6 +123,11 @@ def commit(action, post, authorization = nil)
         end
 
         succeeded = success_from(response)
+
+        response(action, succeeded, response)
+      end
+
+      def response(action, succeeded, response)
```
Comment (bpollack) [inline lib/active_merchant/billing/gateways/checkout_v2.rb]: Nice refactoring! This made the code quite a bit easier to read.

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 2]
```diff
@@ -163,16 +160,16 @@ def base_url
       end
 
       def avs_result(action, succeeded, response)
-        if succeeded
-          action == :purchase ? AVSResult.new(code: response["card"]["avsCheck"]) : nil
+        if succeeded && action == :purchase || :authorize
```
Comment (bpollack) [inline lib/active_merchant/billing/gateways/checkout_v2.rb]: I am pretty sure you meant `action == :purchased || action == :authorized`; as-is, `:authorized` is truthy, so this condition is always true.

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 3]
```diff
@@ -163,16 +160,16 @@ def base_url
...
+          AVSResult.new(code: response['card']['avsCheck']) if response['card'] && response['card']['avsCheck']
```
Comment (bpollack) [inline lib/active_merchant/billing/gateways/checkout_v2.rb]: This is personal preference, but this condition is getting sufficiently complex I'd generally prefer it not to be a guard clause.

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 4]
```diff
@@ -163,16 +160,16 @@ def base_url
...
         else
           nil
         end
       end
 
       def cvv_result(action, succeeded, response)
-        if succeeded
-          action == :purchase ? CVVResult.new(response["card"]["cvvCheck"]) : nil
+        if succeeded && action == :purchase || :authorize
```
Comment (bpollack) [inline lib/active_merchant/billing/gateways/checkout_v2.rb]: Same issue as line [the earlier line](https://github.com/activemerchant/active_merchant/pull/2650/files#diff-390e4ab1e508727fef10daef8ff96d35R163)

File: lib/active_merchant/billing/gateways/checkout_v2.rb (hunk 2 above)
Comment (deedeelavinder) [reply lib/active_merchant/billing/gateways/checkout_v2.rb]: ugh, yes! Thanks for catching.

File: lib/active_merchant/billing/gateways/checkout_v2.rb (hunk 3 above)
Comment (deedeelavinder) [reply lib/active_merchant/billing/gateways/checkout_v2.rb]: Agreed -- removed.

**Pull request closed without merging**